### PR TITLE
Add ability to alter version column in VersionedCollapsingMergeTree.

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Now it's possible to change the type of version column for `VersionedCollapsingMergeTree` with `ALTER` query.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -1458,6 +1458,16 @@ void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, const S
         }
         else if (command.isRequireMutationStage(getInMemoryMetadata()))
         {
+            /// Type change for version column is allowed despite it's a part of sorting key
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:1461]: How we ensure that type modification does not change the order?

Examples:
Int32 -> UInt32
UInt64 -> UInt32
UInt16 -> Int32

Comment (alexey-milovidov) [reply src/Storages/MergeTree/MergeTreeData.cpp:1461]: (there are no tests that check it's prohibited)

**Comment by alexey-milovidov** [comment]
Not a bugfix. Maybe "improvement"?

File: src/Storages/MergeTree/MergeTreeData.cpp (hunk 1 above)
Comment (alesapin) [reply src/Storages/MergeTree/MergeTreeData.cpp:1461]: I don't see any problems with these changes. Yes, merges will work differently, but what can go wrong?

**Comment by alesapin** [comment]
Tests OK, stress failure is known.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Stress test: https://github.com/openssl/openssl/issues/11974

**Pull request merged**